packages/base/src/msg/redepmtion_rate_adapter.rs

```diff
@@ -0,0 +1,44 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Binary, Decimal};
+use cw_ownable::{cw_ownable_execute, cw_ownable_query};
+
+#[cw_ownable_execute]
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateConfig { new_config: Config },
+}
+
+#[cw_ownable_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(Config)]
+    Config {},
+    #[returns(RedemptionRateResponse)]
+    RedemptionRate {
+        denom: String,
+        params: Option<Binary>,
+    },
+}
+
+#[cw_serde]
+pub struct RedemptionRateResponse {
+    pub redemption_rate: Decimal,
+    pub update_time: u64,
+}
+
+#[cw_serde]
+pub struct Config {
+    pub core_contract: String,
```

Comment:
I think we should store core address using `cosmwasm_std::Addr` like we (hopefully) do in each other contract